src/cloudai/_core/registry.py

```
@@ -203,6 +205,52 @@ def update_agent(self, name: str, value: Type[BaseAgent]) -> None:
+
+    def agent_names(self) -> list[str]:
+        """Return all registered agent names, including unresolved entry point agents."""
+        return sorted(set(self.agents_map) | set(self.agent_entrypoints_map))
+
+    def get_agent(self, name: str) -> Type[BaseAgent]:
+        """Resolve an agent class by name, loading entry point agents on first use."""
+        if name in self.agents_map:
+            return self.agents_map[name]
+
+        if name not in self.agent_entrypoints_map:
+            raise KeyError(f"Agent '{name}' not found. Available agents: {self.agent_names()}")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)